modules/core/src/main/java/org/apache/synapse/message/processors/MessageProcessor.java

```diff
@@ -85,4 +85,17 @@
      * @return Name of the file where this artifact is defined
      */
     public String getFileName();
+
+    /**
+     * This method set the target endpoint associated with the message processor. Without a target endpoint
+     * a message processor could not operated successfully.
```

Comment:
Shall we modify the doc comment to reflect it is an optional attribute of the message processor since it is not required for SamplingProcessor?